Dynatrace/dynatrace-operator
# Reconcile Extensions Secret with EEC Token

Follow-up from https://github.com/Dynatrace/dynatrace-operator/pull/3426

## Description

[K8S-10120](https://dt-rnd.atlassian.net/browse/K8S-10120)

New extensions secret is reconciled when `dynakube.spec.extensions.prometheus.enabled = true`
Secret name = `<dk-name>-extensions-token`, containing `eec-token` data, which gets generated only once.

New conditions added related to the secret creation

## How can this be tested?

- Apply dynakube with `dynakube.spec.extensions.prometheus.enabled = true`
- Check that secret gets created
- Check that condition is added
- Check that condition gets deleted if prometheus gets disabled and the secret gets deleted
